### PR TITLE
MOTECH-1954: Added possibility to delete drag and drop bubbles 

### PR DESCRIPTION
**Pull request by tstalka**
- each bubble can be deleted by the "X" button
- fixed the magical disappearance of manipulation
- fixed manipulation dropdown

### CODE REVIEW

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 1]
```diff
@@ -695,7 +718,7 @@
                         trigger: 'manual'
                     }).click(function (event) {
                         event.stopPropagation();
-                        if (!$(this).hasClass('hasPopoverShow')) {
+                        if (!$(this).hasClass('hasPopoverShow') && (event.target === undefined || event.target.className !== 'close badge-close')) {
```
Comment (pgesek) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: why not just do event.target || ...

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 2]
```diff
@@ -705,18 +728,27 @@
                                 otherPopoverElem.removeClass('hasPopoverShow');
                                 otherPopoverElem.removeAttr('ismanipulate');
                             }
-                            if (filter.key) {
+                            if (filter !== undefined && filter.key) {
```
Comment (pgesek) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: why not just do filter && ...

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 3]
```diff
@@ -705,18 +728,27 @@
...
                                 $(this).attr('manipulate', filter.key.split("?").slice(1).join(" "));
                             }
 
                             $(this).addClass('hasPopoverShow');
                             $(this).attr('ismanipulate', 'true');
                             $(this).popover('show');
-                        } else {
+                        } else if (event.target === undefined || event.target.className !== 'close badge-close') {
```
Comment (pgesek) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: same as above

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 4]
```diff
@@ -737,7 +769,7 @@
                     });
 
                     el.on("manipulateChanged", function () {
-                        if (filter.key) {
+                        if (filter !== undefined && filter.key) {
```
Comment (pgesek) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: same as above

**Comment by tstalka** [comment]
test this please

**Pull request merged**